apps/exports/tests/test_scheduling.py

```diff
@@ -0,0 +1,17 @@
+from apps.exports.models import ExportRun
+from apps.exports.tests.test_utils import BaseExportTestCase
+
+
+class TestSchedule(BaseExportTestCase):
+    def test_export_is_scheduled_to_run(self):
+
+        self.assertTrue(self.export_config.is_scheduled_to_run())
+
+        export_run = ExportRun.objects.create(
+            base_export_config=self.export_config,
+        )
+        self.assertTrue(self.export_config.is_scheduled_to_run())
+
+        export_run.status = ExportRun.FAILED
+        export_run.save()
+        self.assertFalse(self.export_config.is_scheduled_to_run())
```

Comment:
Sorry if this is a dumb/annoying question, but can you explain this test? I'm trying to map it to [the code I think it's testing](https://github.com/dimagi/commcare-sync/blob/90982cd67e0353b71980437c3912aea4403629ea/apps/exports/scheduling.py#L5-L19) and I don't totally understand what about setting the status causes it to no longer be scheduled. I'm also kind of confused by line 13 (I would have thought it shouldn't be scheduled anymore if there's already a queued run).

I realize these tests are passing, I'd just like to understand why.

Reply:
Hey, this is a bit confusing! 

I updated this specific test as it was actually going through a different code path than I expected (this was returning `False` because of line 14 in the code you linked). 

The confusing part is that `is_scheduled_to_run` isn't actually the only thing that determines if a new run is created or not. That happens [here](https://github.com/dimagi/commcare-sync/blob/fr%2Ftests-tests-tests/apps/exports/tasks.py#L13), and includes a check if there are queued runs.

Reply:
Ah ha. That *is* confusing. Probably could do with some more thoughtful method names and/or cleanup.